### ISSUE TITLE
[no-unused-vars] False positive on enums

### ISSUE THREAD

**Issue by nikola1970** [typescript-eslint/typescript-eslint#1197]
Hi guys, I am using React with --typescript flag and having problems with decalring and using enums as it reports "no-unused-vars".

My example code, ADD, REMOVE, TOGGLE in declared enum are being underlined and reported as "no-unused-vars":
```JS
enum ActionTypes {
    ADD,
    REMOVE,
    TOGGLE
}

type Action =
    | { type: ActionTypes.ADD; payload: Todo }
    | { type: ActionTypes.REMOVE; payload: string }
    | { type: ActionTypes.TOGGLE; payload: string };
```

.eslint.js:
```JS
{
env: {
        browser: true,
        es6: true
    },
    settings: {
        'import/resolver': {
            node: {
                extensions: ['.js', '.jsx', '.ts', '.tsx']
            }
        }
    },
    extends: ['airbnb'],
    globals: {
        Atomics: 'readonly',
        SharedArrayBuffer: 'readonly'
    },
    parser: '@typescript-eslint/parser',
    parserOptions: {
        ecmaFeatures: {
            jsx: true
        },
        ecmaVersion: 2018,
        sourceType: 'module'
    },
    plugins: ['react', '@typescript-eslint', 'react-hooks'],
    rules: {...}
}
```
package.json:
```JSON
"devDependencies": {
        "@typescript-eslint/eslint-plugin": "^2.7.0",
        "@typescript-eslint/parser": "^2.7.0",
        "eslint": "^6.6.0",
        "eslint-config-airbnb": "^18.0.1",
        "eslint-plugin-import": "^2.18.2",
        "eslint-plugin-jsx-a11y": "^6.2.3",
        "eslint-plugin-react": "^7.16.0",
        "eslint-plugin-react-hooks": "^1.7.0"
    }
```
**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | 2.7.0 |
| `@typescript-eslint/parser`        | 2.7.0 |
| `TypeScript`                       | 3.7.2 |
| `ESLint`                           | ^6.6.0 |
| `node`                             | 11.10.1 |
| `npm`                              | 6.13.0 |

**Comment by bradzacher** [comment]
You cleared out your eslint config for some reason...

Are you using `no-unused-vars` or `@typescript-eslint/no-unused-vars`? If you're using the former - then it's going to break because it doesn't understand typescript.

**Comment by nikola1970** [comment]
@bradzacher No, I do not use that rules at all.

**Comment by bradzacher** [comment]
you need to make sure you're using our version of the rule.
the airbnb ruleset has no idea about typescript rules.
consider using a typescript variant of the airbnb config, or manually overriding where appropriate.

This is listed in the docs.

**Comment by nikola1970** [comment]
@bradzacher Okay, I think I managed to make it work now. Thanks. :)

**Comment by Magmagan** [comment]
> you need to make sure you're using our version of the rule.
> the airbnb ruleset has no idea about typescript rules.
> consider using a typescript variant of the airbnb config, or manually overriding where appropriate.
> 
> **This is listed in the docs.**

Because I came here from a google search, an explicit link to the docs may be helpful for future people with this problem: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars.md#options

**Comment by Nutscracker23** [comment]
@nikola1970 If you will only work with .ts / tsx then try to change: `extends: ['airbnb'],` to `extends: ['airbnb-typescript']` and try to leave only `plugins: ['@typescript-eslint']`.

Today I got the same problem. Not sure if this solves everything I need, but the project started without any problems.

```
{
  env: {
    browser: true,
    es6: true,
  },
  extends: [
    'airbnb-typescript',
  ],
  globals: {
    Atomics: 'readonly',
    SharedArrayBuffer: 'readonly',
  },
  parser: '@typescript-eslint/parser',
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
    ecmaVersion: 2018,
    sourceType: 'module',
    project: "./tsconfig.json",
    tsconfigRootDir: __dirname,
  },
  plugins: [
    '@typescript-eslint',
  ],
  rules: { ... },
}
```

And devDependencies
```
{
  "@typescript-eslint/eslint-plugin": "^2.19.0",
  "eslint": "^6.8.0",
  "eslint-config-airbnb-typescript": "^6.3.1",
  "eslint-loader": "^3.0.3",
  "eslint-plugin-import": "^2.20.1",
  "eslint-plugin-jsx-a11y": "^6.2.3",
  "eslint-plugin-react": "^7.18.3",
  "eslint-plugin-react-hooks": "^2.3.0"
}
```

**Comment by bradzacher** [comment]
I recommend that everyone should read our getting started guide, as these things are covered.

[Getting Started - Next Steps - ESLint Configs](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/README.md#eslint-configs)

and

[FAQ - I am using a rule from ESLint core, and it doesn't work correctly with TypeScript code](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md#i-am-using-a-rule-from-eslint-core-and-it-doesnt-work-correctly-with-typescript-code)

I will be locking this issue, as it has been resolved.
